### ISSUE TITLE
Intermittent test failure due to non-ascii character in the content

### ISSUE THREAD

**Issue by pumazi** [Connexions/cnx-press#140]
This is an occasional failure that happens during testing. I've not been able to reliably reproduce it. I've asked @kerwinso and @Bkblodget to look into publishing unicode content. That should help us get a better idea what if any issue exists before going on a wild goose chase. 

```python
___________________ test_publish_revision_with_new_abstract ____________________
self = <sqlalchemy.engine.base.Connection object at 0x7f9703af1eb8>
dialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7f9704cb4e80>
constructor = <bound method DefaultExecutionContext._init_compiled of <class 'sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2'>>
statement = 'INSERT INTO module_files (module_ident, fileid, filename) VALUES (%(module_ident)s, %(fileid)s, %(filename)s)'
parameters = {'fileid': 109, 'filename': 'index.cnxml', 'module_ident': 73}
args = (<sqlalchemy.dialects.postgresql.psycopg2.PGCompiler_psycopg2 object at 0x7f97037ea2e8>, [])
conn = <sqlalchemy.pool._ConnectionFairy object at 0x7f9703af1080>
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7f97037ea390>
    def _execute_context(self, dialect, constructor,
                         statement, parameters,
                         *args):
        """Create an :class:`.ExecutionContext` and execute, returning
            a :class:`.ResultProxy`."""
    
        try:
            try:
                conn = self.__connection
            except AttributeError:
                # escape "except AttributeError" before revalidating
                # to prevent misleading stacktraces in Py3K
                conn = None
            if conn is None:
                conn = self._revalidate_connection()
    
            context = constructor(dialect, self, conn, *args)
        except BaseException as e:
            self._handle_dbapi_exception(
                e,
                util.text_type(statement), parameters,
                None, None)
    
        if context.compiled:
            context.pre_exec()
    
        cursor, statement, parameters = context.cursor, \
            context.statement, \
            context.parameters
    
        if not context.executemany:
            parameters = parameters[0]
    
        if self._has_events or self.engine._has_events:
            for fn in self.dispatch.before_cursor_execute:
                statement, parameters = \
                    fn(self, cursor, statement, parameters,
                       context, context.executemany)
    
        if self._echo:
            self.engine.logger.info(statement)
            self.engine.logger.info(
                "%r",
                sql_util._repr_params(parameters, batches=10)
            )
    
        evt_handled = False
        try:
            if context.executemany:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_executemany:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_executemany(
                        cursor,
                        statement,
                        parameters,
                        context)
            elif not parameters and context.no_parameters:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute_no_params:
                        if fn(cursor, statement, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute_no_params(
                        cursor,
                        statement,
                        context)
            else:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute(
                        cursor,
                        statement,
                        parameters,
>                       context)
/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py:1193: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7f9704cb4e80>
cursor = <cursor object at 0x7f97032af900; closed: -1>
statement = 'INSERT INTO module_files (module_ident, fileid, filename) VALUES (%(module_ident)s, %(fileid)s, %(filename)s)'
parameters = {'fileid': 109, 'filename': 'index.cnxml', 'module_ident': 73}
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7f97037ea390>
    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       psycopg2.InternalError: UnicodeEncodeError: 'ascii' codec can't encode character u'\xf4' in position 1: ordinal not in range(128)
E       CONTEXT:  Traceback (most recent call last):
E         PL/Python function "add_module_file", line 3, in <module>
E           return add_module_file(plpy, TD)
E         PL/Python function "add_module_file", line 758, in add_module_file
E         PL/Python function "add_module_file", line 221, in produce_html_for_module
E         PL/Python function "add_module_file", line 296, in produce_transformed_file
E         PL/Python function "add_module_file", line 304, in resolve_urls
E         PL/Python function "add_module_file", line 297, in __call__
E         PL/Python function "add_module_file", line 511, in fix_anchor_references
E         PL/Python function "add_module_file", line 183, in __init__
E         PL/Python function "add_module_file", line 170, in __init__
E       PL/Python function "add_module_file"
/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/default.py:509: InternalError
```

**Comment by Bkblodget** [comment]
I had the content interns find instances of unicode in a few books, unfortunately mostly derived-copies use these, but there are a few instances in non-derived copy books:

https://docs.google.com/spreadsheets/d/1R-w9VlMgYsky409r6qFIXjORJ0FxHSvrLF0-_Xh2bqk/edit#gid=0

**Comment by kerwinso** [comment]
Based on the Google Sheet that Bey created, I located a module in Precalculus (col11667) with unicode content (i.e., a `≠` character). Then, I tried a simple neb publish ("soccer" to "futbol") to that collection, which succeeded: https://staging.cnx.org/contents/_VPq4foj@8.1:UIpNTgE2@4/Matrices-and-Matrix-Operations

More details in TestRail: https://openstax.testrail.com/index.php?/tests/view/346125

Since it looks like Batch is able to successfully publish content with unicode in it, I am closing this issue.

Note: I wasn't able to test using Katalyst University Physics, because those modules are encoded with `cnxml-version=0.8` whereas neb requires `cnxml-version=0.7`.
